wire-ios-system/Tests/ExpiringActivityTests.swift

```diff
@@ -0,0 +1,116 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import XCTest
+@testable import WireSystem
+
+class MockExpiringActivityAPI: ExpiringActivityInterface {
+
+    typealias MethodCall = (_ reason: String, _ block: @escaping @Sendable (Bool) -> Void) -> Void
+
+    var method: MethodCall?
+
+    func performExpiringActivity(withReason reason: String, using block: @escaping @Sendable (Bool) -> Void) {
+        if let method {
+            method(reason, block)
+        } else {
+            fatalError("no mock for `performExpiringActivity(withReason:using:)`")
+        }
+    }
+
+}
+
+class ExpiringActivityTests: XCTestCase {
+
+    let concurrentQueue = DispatchQueue(label: "activity queue", attributes: [.concurrent])
+
+    func testThatTaskIsCancelled_WhenActivityExpires() async throws {
+
+        // given
+        let api = MockExpiringActivityAPI()
```

Comment:
Do I understand correctly that this mock compared to ProcessInfo just execute the method directly instead of waiting 30 sec?

Reply:
The individual test cases determinate how it behave, but yes we execute directly since we don't want to wait for 30 sec.